src/Tokenizer/Tokens.php

```diff
@@ -1110,7 +1110,21 @@ public function setCode(string $code): void
         }
 
         $this->updateSizeToZero(); // clear memory
-        $tokens = token_get_all($code, \TOKEN_PARSE);
+
+        $prevErrorHandler = set_error_handler(static function ($type, $msg, $file, $line, $context = []) use (&$prevErrorHandler) {
```

Comment:
inspired by symfony, where this approach is used in several places. E.g., here: https://github.com/symfony/symfony/blob/32b347f826895b469b7759527d4b2f8663090797/src/Symfony/Component/Config/ResourceCheckerConfigCache.php#L154-L160